lib/pure/json.nim

```diff
@@ -677,7 +685,7 @@ proc pretty*(node: JsonNode, indent = 2): string =
   ## Similar to prettyprint in Python.
   runnableExamples:
     let j = %* {"name": "Isaac", "books": ["Robot Dreams"],
-                "details": {"age": 35, "pi": 3.1415}}
+                "details": {"age": 35, "number": 3.125}}
```

Comment:
I'll repeat the same question/concern as in the original PR (which is now reverted):

Why do we need to change this number? What would it be printed for `"pi": 3.1415`?

Reply:
> What would have been printed for "pi": 3.1415?

pretty much like what i wrote in top post:
```nim
import json
echo 3.1415
echo(%* 3.1415)

3.1415 # this doesn't change (unlike with other PR)
3.1415000000000002 # this changes to fix roundtrip issue
```

> Why do we need to change this number? 

that's to fix roundtrip errors  (pretty much what #13196 is about)
And I've changed the test to avoid showing 3.1415000000000002 in the output because I'm not sure `$(%* 3.1415) = 3.1415000000000002` should be part of the spec (which is more or less what runnableExamples are); this is subject to stdlib changes pending better float to string algorithms such as ryu, see https://github.com/nim-lang/Nim/issues/13365

Reply:
Thanks for the clarification!